apps/web/src/components/layout/AppLayout.tsx

```diff
@@ -63,16 +64,18 @@ export function AppLayout() {
             >
               <SpotLight>
                 <AppShell>
-                  {isInformationArchitectureEnabled ? <MainNav /> : <SideNav />}
-                  <ContentShell>
-                    <FreeTrialBanner />
-                    {isInformationArchitectureEnabled ? (
-                      <HeaderNavNew />
-                    ) : (
-                      <HeaderNav isIntercomOpened={isIntercomOpened} />
-                    )}
-                    <Outlet />
-                  </ContentShell>
+                  <DocsProvider>
```

Comment:
<!-- Issues highlight specific problems with the subject under review. These problems can be user-facing or behind the scenes. It is strongly recommended to pair this comment with a suggestion. If you are not sure if a problem exists or not, consider leaving a question.  -->
⚠ **issue:** I'm concerned that a context wrapper around the whole application here could be problematic as `Context` doesn't provide any memorization out of the box.‏